heralding/libs/msrdp/parser.py

```diff
@@ -0,0 +1,278 @@
+# Copyright (C) 2019 Sudipta Pandit <realsdx@protonmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import struct
+
+
+class InvalidExpectedData(Exception):
+    def __init__(self, message=""):
+        Exception.__init__(self, msg)
+
+
+class RawBytes():
+    """ Read/Consume raw bytes """
+
+    def __init__(self, data, structFormat, typeSize, pos=0, optional=False):
+        self.data = data
+        self._pos = pos
+        self._structFormat = structFormat
+        self._typeSize = typeSize
+        self._optional = optional
+
+    def dataLen(self):
+        return len(self.data[self._pos:])
+
+    def read(self):
+        if self.dataLen() < self._typeSize:
+            if self._optional:
+                print("No optional data present")
```

Comment:
more print statements. If these are needed for debugging purposes, please use logger...

Reply:
Sorry, i totally forgot about these